### PR TITLE
Document virtual environment discovery

### PR DESCRIPTION
**Pull request by zanieb**

### CODE REVIEW

**Comment by zanieb** [comment]
I'm not entirely sure where this should live

**Comment by zanieb** [comment]
We might need a virtual environment concept document

File: docs/concepts/python-versions.md [hunk 1]
```diff
@@ -134,6 +134,20 @@ To exclude downloads and only show installed Python versions:
 $ uv python list --only-installed
 ```
 
+## Discovery of virtual environments
+
+Some uv commands may use a Python interpreter from a virtual environment. When searching for virtual
+environments, uv uses a virtual environment from:
```
Comment (charliermarsh) [inline docs/concepts/python-versions.md:140]: "uv prioritizes (in order):"

File: docs/concepts/python-versions.md [hunk 2]
```diff
@@ -134,6 +134,20 @@ To exclude downloads and only show installed Python versions:
...
+
+- The `VIRTUAL_ENV` environment variable.
+- The `CONDA_PREFIX` environment variable.
+- A `.venv` directory in the working directory.
+- A `.venv` directory in any parent directory.
+
+After exhausting these possibilities, uv will either
+[search for a Python installation](#discovery-of-python-versions) or exit with an error if the
+command requires using a virtual environment.
```
Comment (charliermarsh) [inline docs/concepts/python-versions.md:149]: "requires a virtual environment"

**Review by charliermarsh — APPROVED** [review]

File: docs/concepts/python-versions.md (hunk 2 above)
Comment (zanieb) [reply docs/concepts/python-versions.md:149]:
```suggestion
command requires a virtual environment.
```

File: docs/concepts/python-versions.md (hunk 1 above)
Comment (zanieb) [reply docs/concepts/python-versions.md:140]:
```suggestion
environments, uv prioritizes (in order):
```

**Pull request merged**